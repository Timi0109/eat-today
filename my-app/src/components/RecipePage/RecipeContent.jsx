import React, {useState} from "react";
import { Link,useLocation } from 'react-router-dom';
import styles from "./RecipeContent.module.css";
import TextField from '@material-ui/core/TextField';
import {withStyles } from "@material-ui/core/styles";
import { pink } from '@material-ui/core/colors';

const RecipeContent = ({food, recipes}) => {
  
const CssTextField = withStyles({
    root: {
        '& .MuiInput-underline:before':{
            borderBottomColor: "#ea86b6",
            "&:hover": {
                borderBottomColor: "#ea86b6",
            },
        },
        '& label.Mui-focused': {
            color: "#ea86b6"
        },
        '& .MuiInput-underline:after': {
            borderBottomColor: "#ea86b6",
        }
        
    },
   
  })(TextField);

    const meat = food.Meat !== undefined ?food.Meat.toLowerCase() :null;
    const veggies = food.Veggies !== undefined ?food.Veggies.toLowerCase() :null;
    const others = food.Others !== undefined ?food.Others.toLowerCase() :null;
    const [filterName, setFilterName] = useState(null);
    const [detail, setDetail] = useState(false);

    let filterRecipe;
    if(meat === null && veggies === null && others === null){
        filterRecipe = recipes.slice(0);
    }else{
        filterRecipe = recipes.filter(x =>  x.name.toLowerCase().includes(meat) || x.name.toLowerCase().includes(veggies) ||
                                        x.name.toLowerCase().includes(others) || (x.meat !== undefined && x.meat.toLowerCase().includes(meat)) ||
                                        (x.veggies !== undefined && x.veggies.toLowerCase().includes(veggies)) || (x.others !== undefined && x.others.toLowerCase().includes(others)));
    }

    if(filterName !== null){
        filterRecipe = filterRecipe.filter((x) => x.name.toLowerCase().includes(filterName.toLowerCase()) ||
                                            (x.meat !== undefined && x.meat.toLowerCase().includes(filterName.toLowerCase())) ||
                                            (x.veggies !== undefined && x.veggies.toLowerCase().includes(filterName.toLowerCase())) || 
                                            (x.others !== undefined && x.others.toLowerCase().includes(filterName.toLowerCase())));
    }

    return (
        <div className={styles.contentContainer}>
              
                <div className={styles.searchContainer}> 
                    {/* <CssTextField  className={styles.searchInput} id="custom-css-standard-input" variant="standard" label="Search by name" onChange={(e) => {setFilterName(e.target.value)}}></CssTextField> */}
                    {/* <CssTextField className={styles.searchInput} label="Search by name" variant="outlined"  onChange={(e) => {setFilterName(e.target.value)}}/> */}
                    <input className={styles.searchInput} placeholder="Search by name" onChange={e => {setFilterName(e.target.value)}}></input>
                </div>
            
                <div className={styles.listContainer}>
                    <div className={styles.outerRecipeContainer} >
                    {filterRecipe.map((item, id)=> {
                        return (
                            <div className={styles.interListContainer} key={id}>
                                <div className={styles.recipeContainer} >
                                    <Link className={styles.link} to={{pathname: `/recipe-list/${item.name}`, state: {item}}}>
                                        <img className={styles.recipeImg} src={item.url} alt="recipe img"></img>
                                    </Link>
                                    <Link className={styles.link} to={{pathname: `/recipe-list/${item.name}`, state: {item}}}>
                                        <div className={styles.title}>
                                            <p className={styles.recipeName}>{item.name}</p>
                                            {item.meat ?<p className={styles.meat}>Meat: {item.meat}</p> :null}
                                            {item.veggies ?<p className={styles.veggies}>Veggies: {item.veggies}</p> :null}
                                        </div>
                                    </Link>
                                    
                                </div>
                                <hr></hr>
                            </div>
                        )
                    })}
                    </div>
                </div>
        </div>
    )
}

export default RecipeContent