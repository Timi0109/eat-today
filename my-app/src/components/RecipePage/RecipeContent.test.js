import React from "react";
import RecipeContent from "./RecipeContent";

describe("RecipeContent tests", () => {
  it("renders correctly", () => {
      const r = 
           [ {
              "name":"Beef and Broccoli Lo Mein",
              "meat":"Beef",
              "veggies":"Brocoli",
              "others":"Pasta",
              "url":"https://img.buzzfeed.com/buzzfeed-static/static/2020-04/7/14/asset/af2b435023ab/sub-buzz-265-1586268288-7.jpg",
              "instruction":[
                 "Cook pasta: In a large pot with boiling water, cook the noodles according to package directions. Add the broccoli the last 5 minutes of cooking and let them cook until tender. Drain the noodles and broccoli.",
                 "Cook steak: While the pasta is cooking, add olive oil to a medium sized skillet. Cook the steak until no longer pink. Add the garlic, and carrots and cook for a minute more.",
                 "Mix sauce: In a small bowl whisk together the brown sugar soy sauce, hoisin sauce, sesame oil, ginger, red pepper and pepper.",
                 "Toss everything together: Add the spaghetti to the skillet and pour the sauce on top and toss until incorporated."
              ]}]
           
    const { container } = render(<RecipeContent food={{}} recipes={r}></RecipeContent>);
    expect(container).toMatchSnapshot();
  });

});
