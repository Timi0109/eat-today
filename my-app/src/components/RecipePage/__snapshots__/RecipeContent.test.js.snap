// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeContent tests renders correctly 1`] = `
<div>
  <div
    class="contentContainer"
  >
    <input
      class="searchInput"
      placeholder="Search by name"
    />
    <hr />
    <div
      class="listContainer"
    >
      <div
        class="outerRecipeContainer"
      >
        <div
          class="interListContainer"
        >
          <div
            class="recipeContainer"
          >
            <a
              class="link"
              href="/recipe-list/Beef and Broccoli Lo Mein"
            >
              <img
                alt="recipe img"
                class="recipeImg"
                src="https://img.buzzfeed.com/buzzfeed-static/static/2020-04/7/14/asset/af2b435023ab/sub-buzz-265-1586268288-7.jpg"
              />
            </a>
            <a
              class="link"
              href="/recipe-list/Beef and Broccoli Lo Mein"
            >
              <div
                class="title"
              >
                <p
                  class="recipeName"
                >
                  Beef and Broccoli Lo Mein
                </p>
                <p
                  class="meat"
                >
                  Meat: 
                  Beef
                </p>
                <p
                  class="veggies"
                >
                  Veggies: 
                  Brocoli
                </p>
              </div>
            </a>
          </div>
          <hr />
        </div>
      </div>
    </div>
  </div>
</div>
`;
